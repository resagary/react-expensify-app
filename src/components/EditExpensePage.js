import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    state = {
        selectedOption: false
    };

    openModal = () => {
        this.setState(() => ({ selectedOption: true }));
    };

    closeModal = () => {
        this.setState(() => ({ selectedOption: false }));
    };

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    };

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                      expense={this.props.expense}
                      onSubmit={this.onSubmit}
                    />
                    <button
                      className="button button--secondary"
                      onClick={this.openModal}
                    > Remove
                    </button>
                </div>
                <Modal
                  isOpen={this.state.selectedOption}
                  onRequestClose={this.closeModal}
                  contentLabel="Remove Expense"
                  className="modal"
                  closeTimeoutMS={200}
                  appElement={document.getElementById('app')}
                >
                    <h3 className="modal__title">Remove Expense</h3>
                    <p className="modal__body">Are you sure you want to remove {this.props.expense.description}?</p>
                    <button className="button--remove" onClick={this.onRemove}>Yes</button>
                    <button className="button button--secondary" onClick={this.closeModal}>No</button>
                </Modal>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);