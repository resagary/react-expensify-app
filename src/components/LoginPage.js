import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startEmailLogin, startGoogleLogin } from '../actions/auth';

export class LoginPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            email: '',
            password: '',
            error: ''
        }
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.email || !this.state.password) {
            this.setState(() => ({ error: 'Please provide an email, username and password.'}));
        } else {
            this.setState(() => ({ error:'' }));
            this.props.startEmailLogin(this.state.email, this.state.password).then(() => {
                console.log('User logged in.');
            }).catch((error) => {
                if (error.code === 'auth/user-not-found') {
                    this.setState(() => ({ error: 'There is no account with the email you\'ve entered.' }));
                }
                if (error.code === 'auth/wrong-password') {
                    this.setState(() => ({ error: 'The email or password is incorrect.' }));
                }
            });
        }
    };

    render () {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify</h1>
                    <p>Keep track of your expenses</p>
                    <form  id="emailLogin" className="form" onSubmit={this.onSubmit}>
                        <input
                          className="text-input"
                          type="text"
                          placeholder="Email"
                          autoFocus
                          value={this.state.email}
                          onChange={this.onEmailChange}
                        />
                        <input
                          className="text-input"
                          type="password"
                          placeholder="Password"
                          autoFocus
                          value={this.state.password}
                          onChange={this.onPasswordChange}
                        />
                        <div>
                            <button className="button">Login</button>
                        </div>
                        <div>
                            <Link className="button--forgotpw button--link" to="/reset-pw">
                              Forgot Password?
                            </Link>
                        </div>
                    </form>
                    <p>or login with</p>
                    <button
                      id="googleLogin" 
                      className="button-round" 
                      onClick={this.props.startGoogleLogin}
                    ><img src="https://img.icons8.com/color/48/000000/google-logo.png"/>
                    </button>
                    <Link className="button button-text button--link" to="/sign-up">
                      Don't have an account? Sign Up
                    </Link>
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startEmailLogin: (email, password) => dispatch(startEmailLogin(email, password))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);