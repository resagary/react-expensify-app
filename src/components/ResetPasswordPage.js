import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startPasswordReset } from '../actions/auth';

export class ResetPasswordPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            email: '',
            error: ''
        };
    }

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    };

    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.email) {
            this.setState(() => ({ error: 'Please provide an email.'}));
        } else {
            this.setState(() => ({ error:'' }));
            this.props.startPasswordReset(this.state.email).then(() => {
                this.props.history.push('/');
            });
        }
    };

    render () {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify</h1>
                    <h3 className="box-layout__subtitle">Reset Password</h3>
                    <form id="resetPasswordForm" className="form" onSubmit={this.onSubmit}>
                        {this.state.error && <p className="form__error">{this.state.error}</p>}
                        <input
                            className="text-input"
                            type="text"
                            placeholder="Email"
                            autoFocus
                            value={this.state.email}
                            onChange={this.onEmailChange}
                        />
                        <div>
                            <button className="button">Reset</button>
                        </div>
                    </form>
                    <Link className="button button-text button--link" to="/">
                      Sign In
                    </Link>
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch, props) => ({
    startPasswordReset: (email) => dispatch(startPasswordReset(email))
});

export default connect(undefined, mapDispatchToProps)(ResetPasswordPage);