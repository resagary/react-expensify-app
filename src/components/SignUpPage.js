import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startSignUpUser } from '../actions/auth';

export class SignUpPage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            username: '',
            email: '',
            password: '',
            confirmPassword: '',
            error: ''
        };
    }

    onUsernameChange = (e) => {
        const username = e.target.value;
        this.setState(() => ({ username }));
    };

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    };

    onConfirmPasswordChange = (e) => {
        const confirmPassword = e.target.value;
        this.setState(() => ({ confirmPassword }));
    };

    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.username || 
            !this.state.email || 
            !this.state.password || 
            !this.state.confirmPassword
        ) {
            this.setState(() => ({ error: 'Please provide an email, username and password.'}));
        } else if (this.state.password !== this.state.confirmPassword) {
            this.setState(() => ({ error: 'Passwords do not match.'}));
        } else {
            this.setState(() => ({ error:'' }));
            this.props.startSignUpUser(this.state.email, this.state.password);
        }
    };

    render () {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify</h1>
                    <h3 className="box-layout__subtitle">Sign Up</h3>
                    <form id="signUpForm" className="form" onSubmit={this.onSubmit}>
                        {this.state.error && <p className="form__error">{this.state.error}</p>}
                        <input
                            className="text-input"
                            type="text"
                            placeholder="Full Name"
                            autoFocus
                            value={this.state.username}
                            onChange={this.onUsernameChange}
                        />
                        <input
                            className="text-input"
                            type="text"
                            placeholder="Email"
                            autoFocus
                            value={this.state.email}
                            onChange={this.onEmailChange}
                        />
                        <input
                            className="text-input"
                            type="password"
                            placeholder="Password"
                            autoFocus
                            value={this.state.password}
                            onChange={this.onPasswordChange}
                        />
                        <input
                            className="text-input"
                            type="password"
                            placeholder="Confirm Password"
                            autoFocus
                            value={this.state.confirmPassword}
                            onChange={this.onConfirmPasswordChange}
                        />
                        <div>
                            <button className="button">Sign Up</button>
                        </div>
                    </form>
                    <Link className="button button-text button--link" to="/">
                      Already a user? Sign In
                    </Link>
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch, props) => ({
    startSignUpUser: (email, password) => dispatch(startSignUpUser(email, password))
});

export default connect(undefined, mapDispatchToProps)(SignUpPage);