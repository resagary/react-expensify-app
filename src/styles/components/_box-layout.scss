.box-layout {
    align-items: center;
    background: url('/images/coinstack-bg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box {
    background: fade-out($color: white, $amount: .15);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 35rem;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
}

.box-layout__subtitle {
    margin: 0 0 $s-size 0;
    line-height: 1;
}