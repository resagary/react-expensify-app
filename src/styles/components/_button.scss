.button {
    background: $blue;
    border: none;
    color: white;
    display: inline-block;
    font-size: $font-size-large;
    font-weight: 300;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button-round {
    background: none;
    border: 1px solid $light-gray;
    border-radius: 50%;
    padding: $m-size;
}

.button-text {
    color: $gray;
    display: inline-block;
    text-decoration: none;
}

.button-text:hover {
    color: $dark-gray;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #990000;
    margin-bottom: 20px;
}