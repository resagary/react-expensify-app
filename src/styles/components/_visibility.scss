.show-for-desktop {
    @media (min-width: $desktop-breakpoint - 0.1rem) {
        display: none;
    }
}

.show-for-mobile {
    @media (min-width: $desktop-breakpoint) {
        display: none;
    }
}