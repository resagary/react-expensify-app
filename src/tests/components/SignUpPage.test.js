import React from 'react';
import { shallow } from 'enzyme';
import { SignUpPage } from '../../components/SignUpPage';

let startSignUpUser, wrapper;

beforeEach(() => {
    //startSignUpUser = jest.fn();
    wrapper = shallow(
        <SignUpPage 
          //startSignUpUser={startSignUpUser}
        />
    );
});

test('should correctly render SignUpPage', () => {
    expect(wrapper).toMatchSnapshot();
});

// test('should handle startSignUpUser on form submit', () => {
//     const email = 'testing1@yahoo.com';
//     const password = '9876zyx';
//     wrapper.find('#signUpForm').prop('onSubmit')(email, password);
//     expect(startSignUpUser).toHaveBeenCalledWith(email, password);
// });