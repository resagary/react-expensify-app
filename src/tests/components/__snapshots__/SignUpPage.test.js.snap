// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render SignUpPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <h3
      className="box-layout__subtitle"
    >
      Sign Up
    </h3>
    <form
      className="form"
      id="signUpForm"
      onSubmit={[Function]}
    >
      <input
        autoFocus={true}
        className="text-input"
        onChange={[Function]}
        placeholder="Full Name"
        type="text"
        value=""
      />
      <input
        autoFocus={true}
        className="text-input"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
      <input
        autoFocus={true}
        className="text-input"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <input
        autoFocus={true}
        className="text-input"
        onChange={[Function]}
        placeholder="Confirm Password"
        type="password"
        value=""
      />
      <div>
        <button
          className="button"
        >
          Sign Up
        </button>
      </div>
    </form>
    <Link
      className="button button-text button--link"
      replace={false}
      to="/"
    >
      Already a user? Sign In
    </Link>
  </div>
</div>
`;
